File: spring-boot-kafka-demo/kafka-json-demo1/src/main/java/com/github/jccode/springbootkafkademo/jsondemo1/Application.java
package com.github.jccode.springbootkafkademo.jsondemo1;


import com.github.jccode.springbootkafkademo.jsondemo1.config.JsonConfig;
import com.github.jccode.springbootkafkademo.jsondemo1.event.EventProducer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Import;

@SpringBootApplication
@Import({JsonConfig.class})
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args).close();
    }

    @Autowired
    private EventProducer producer;

    @Bean
    public CommandLineRunner eventTestRunner(ApplicationContext ctx) {
        return args -> {
            producer.produceEvents1();
            producer.produceEvents2();
        };
    }
}

File: spring-boot-kafka-demo/kafka-json-demo1/src/main/scala/com/github/jccode/springbootkafkademo/jsondemo1/Application.scala
package com.github.jccode.springbootkafkademo.jsondemo1

import com.github.jccode.springbootkafkademo.jsondemo1.config.{CommonConfig, JsonConfig}
import com.github.jccode.springbootkafkademo.jsondemo1.event.EventProducer
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.{CommandLineRunner, SpringApplication}
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.{Bean, Import}

@SpringBootApplication
@Import(Array(classOf[JsonConfig]))
class Application {

  @Autowired private val producer: EventProducer = null

  /*
  @Bean def eventTestRunner(ctx: ApplicationContext): CommandLineRunner = (args: Array[String]) => {
    def foo(args: Array[String]) = {
      producer.produceEvents1
      producer.produceEvents2
    }

    foo(args)
  }
  */

  @Bean
  def eventTestRunner(ctx: ApplicationContext) = new CommandLineRunner {
    override def run(strings: String*): Unit = {
      producer.produceEvents1()
      producer.produceEvents2()
    }
  }
}

object Application {
  def main(args: Array[String]): Unit = {
    SpringApplication.run(classOf[Application], args: _*)
  }
}

